targets/TARGET_Cypress/TARGET_PSOC6/xip_user.c

```diff
@@ -0,0 +1,84 @@
+/* mbed Microcontroller Library
```

Comment:
there is a code example for setting this up:
https://github.com/cypresssemiconductorco/mbed-os-example-xip

not sure it should be done here

Reply:
XIP mode has to be enabled prior to constructors being called. If we were using C not C++, the example would have worked. Unfortunately the Mbed OS is mostly C++ based.

Reply:
At what points does this data need to be accessible?  We have some updates in mind that would use some of the memory for data storage which would need to have XIP mode turned off.

Reply:
XIP needs to be set up prior to mbed_toolchain_init. See the following link for details.
https://os.mbed.com/docs/mbed-os/v5.11/feature-hal-spec-usb-device-doxy/group__boot.html

Reply:
Is it safe to disable after mbed_sdk_init() finishes?

Reply:
In theory, I would think so as long as no function in the XIP region is called afterwards, although I haven't tested it. I'll do more testing on dynamic switch and sharing between storage and execution once the QSPI fix is in. Currently storage functionality can't be tested.